planner/core/src/main/java/eu/seaclouds/platform/planner/core/DamGenerator.java

```diff
@@ -272,6 +283,27 @@ public static String generateDam(String adp, String monitorGenURL, String monito
         return adpYaml;
     }
 
+    public static Map<String, Object> translateOfferingsType(Map<String, Object> adpYaml) {
+
+        Map<String, Object> topologyTemplate = (Map<String, Object>) adpYaml.get(TOPOLOGY_TEMPLATE);
+        Map<String, Object> nodeTemplates = (Map<String, Object>) topologyTemplate.get(NODE_TEMPLATES);
+
+        for(String moduleName:nodeTemplates.keySet()) {
+            Map<String, Object> module = (Map<String, Object>) nodeTemplates.get(moduleName);
+
+            String moduleType = (String) module.get("type");
+
+            // replaces seaclouds types (unknown by the deployer) with tosca types for the offerings
+            if (moduleType.startsWith("seaclouds.nodes.Platform.")) {
```

Comment:
Yes, you are right. This type come out from the offerings fetched by the Discoverer. I am trying to understand if it is feasible to change that type in the very first step. In that case this part of the code will be no more needed.